src/k-provider/ovp/provider.js

```diff
@@ -74,16 +88,25 @@ export default class OVPProvider extends BaseProvider<OVPProviderMediaInfoObject
     });
   }
 
-  doRequest(loaders: Array<RequestLoader>): Promise<any> {
-    const dataLoader = new OVPDataLoaderManager(this.playerVersion, this.partnerId, this.ks, this._networkRetryConfig);
+  doRequest(loaders: Array<RequestLoader>, ks?: string): Promise<any> {
+    if (ks) {
+      this.ks = ks;
+      this._isAnonymous = false;
+    }
+    const dataLoader = new OVPDataLoaderManager(this.playerVersion, this.partnerId, this.userKs, this._networkRetryConfig);
 
     return new Promise((resolve, reject) => {
+      if (!this.userKs) {
+        dataLoader.add(OVPSessionLoader, {widgetId: this.widgetId});
+      }
       loaders.forEach((loaderRequest: RequestLoader) => {
+        if (!this.userKs) loaderRequest.params.ks = '{1:result:ks}';
```

Comment:
```suggestion
        loaderRequest.params.ks = this.userKs || '{1:result:ks}';
```
so each loader gets and passes the ks either real or token (as we do in getMediaConfig)